sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DoNotHardcodeCredentialsBase.cs

```diff
@@ -148,13 +151,30 @@ private IEnumerable<string> FindCredentialWords(string variableName, string vari
                 var matches = analyzer.passwordValuePattern.Matches(variableValue);
                 foreach (Match match in matches)
```

Comment:
Can we add test to prove this `foreach` usefull? That would require multiple passwords in single string situation and that does not make sense and is not handled by current regex.
I guess we can't due to current regex.
So I suggest to use `.Match` and use values directly. Check for `match.Success` should be enough.